From: Tobias Frost <tobi@debian.org>
Date: Sun, 15 May 2016 21:03:09 +0200
Subject: fix_hurd_build

posix_fadvise64 is not available on hurd, will yield to an compiler warning and
aborting the build through --fatal-warnings

Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=799619
Forwarded: no
---
 src/ioslaves/file/file.cpp      | 2 +-
 src/ioslaves/file/file_unix.cpp | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

Index: kio/src/ioslaves/file/file.cpp
===================================================================
--- kio.orig/src/ioslaves/file/file.cpp
+++ kio/src/ioslaves/file/file.cpp
@@ -342,7 +342,7 @@ void FileProtocol::get(const QUrl &url)
         return;
     }
 
-#if HAVE_FADVISE
+#if HAVE_FADVISE && !defined(__GNU__)
     //TODO check return code
     posix_fadvise(f.handle(), 0, 0, POSIX_FADV_SEQUENTIAL);
 #endif
Index: kio/src/ioslaves/file/file_unix.cpp
===================================================================
--- kio.orig/src/ioslaves/file/file_unix.cpp
+++ kio/src/ioslaves/file/file_unix.cpp
@@ -128,7 +128,7 @@ void FileProtocol::copy(const QUrl &srcU
         return;
     }
 
-#if HAVE_FADVISE
+#if HAVE_FADVISE && !defined(__GNU__)
     posix_fadvise(src_file.handle(), 0, 0, POSIX_FADV_SEQUENTIAL);
 #endif
 
@@ -147,7 +147,7 @@ void FileProtocol::copy(const QUrl &srcU
     // nobody shall be allowed to peek into the file during creation
     dest_file.setPermissions(QFileDevice::ReadOwner | QFileDevice::WriteOwner);
 
-#if HAVE_FADVISE
+#if HAVE_FADVISE && !defined(__GNU__)
     posix_fadvise(dest_file.handle(), 0, 0, POSIX_FADV_SEQUENTIAL);
 #endif
 
